### PR TITLE
Add argminmax operator

### PR DESCRIPTION
**Pull request by tmartin-gh**

### CODE REVIEW

File: include/matx/operators/argminmax.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+      template <typename Out, typename Executor>
+      void Exec(Out &&out, Executor &&ex) const {
+        static_assert(cuda::std::tuple_size_v<remove_cvref_t<Out>> == 5, "Must use mtie with 4 outputs on argminmax(). ie: (mtie(MinVal, MinIdx, MaxVal, MaxIdx) = argminmax(A))");   
+        argminmax_impl(cuda::std::get<0>(out), cuda::std::get<1>(out), cuda::std::get<2>(out), cuda::std::get<3>(out), a_, ex);
+      }
+
+      static __MATX_INLINE__ constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank()
+      {
+        return matxNoRank;
```
Comment (cliffburdick) [inline include/matx/operators/argminmax.h:74]: Shouldn't this have a rank of all the outputs?

File: include/matx/transforms/cub.h [hunk 2]
```diff
@@ -76,6 +76,7 @@ typedef enum {
   CUB_OP_SELECT_IDX,
   CUB_OP_UNIQUE,
   CUB_OP_ARG_REDUCE,
+  CUB_OP_DUAL_ARG_REDUCE,
```
Comment (cliffburdick) [inline include/matx/transforms/cub.h:79]: Quad?

File: include/matx/transforms/reduce.h [hunk 3]
```diff
@@ -2234,6 +2234,52 @@ void __MATX_INLINE__ argmin_impl(OutType dest, TensorIndexType &idest, const InT
 #endif
 }
 
+/**
+ * Compute min and max reduction of a tensor and returns value + index
```
Comment (cliffburdick) [inline include/matx/transforms/reduce.h:2238]: Of an operator

**Review by cliffburdick — APPROVED** [review]

**Review by cliffburdick — CHANGES_REQUESTED** [review]
Needs unit tests

**Comment by cliffburdick** [comment]
@tylera-nvidia please look at this one and test the performance if you still have that data.

File: include/matx/transforms/cub.h (hunk 2 above)
Comment (tmartin-gh) [reply include/matx/transforms/cub.h:79]: Changed to CUB_OP_SINGLE_ARG_REDUCE and CUB_OP_DUAL_ARG_REDUCE

File: include/matx/transforms/reduce.h (hunk 3 above)
Comment (tmartin-gh) [reply include/matx/transforms/reduce.h:2238]: Fixed for argminmax, argmax, and argmin.

**Comment by tmartin-gh** [comment]
> Needs unit tests and documentation

Added unit tests.

File: include/matx/operators/argminmax.h (hunk 1 above)
Comment (tmartin-gh) [reply include/matx/operators/argminmax.h]: Copy bug from argmin.  Fixed for argmin, argmax, and argminmax.

**Comment by tmartin-gh** [comment]
> Needs unit tests and documentation

Added documentation for argminmax

**Review by cliffburdick — APPROVED** [review]

**Pull request merged**